packages/core/lib/commands/call/run.js

```diff
@@ -0,0 +1,167 @@
+module.exports = async function (options) {
+  const fs = require("fs");
+  const util = require("util");
+  const { Environment } = require("@truffle/environment");
+  const OS = require("os");
+  const Codec = require("@truffle/codec");
+  const Encoder = require("@truffle/encoder");
+  const Decoder = require("@truffle/decoder");
+  const TruffleError = require("@truffle/error");
+  const { fetchAndCompile } = require("@truffle/fetch-and-compile");
+  const loadConfig = require("../../loadConfig");
+
+  if (options.url && options.network) {
+    const message =
+      "" +
+      "Mutually exclusive options, --url and --network detected!" +
+      OS.EOL +
+      "Please use either --url or --network and try again." +
+      OS.EOL +
+      "See: https://trufflesuite.com/docs/truffle/reference/truffle-commands/#call" +
+      OS.EOL;
+    throw new TruffleError(message);
+  }
+
+  let config = loadConfig(options);
+  await Environment.detect(config);
+
+  const [contractNameOrAddress, functionNameOrSignature, ...args] = config._;
+  let functionEntry, transaction;
+
+  const { encoder, decoder } = config.fetchExternal
+    ? await sourceFromExternal(contractNameOrAddress, config)
+    : await sourceFromLocal(contractNameOrAddress, config);
+
+  ({ abi: functionEntry, tx: transaction } = await encoder.encodeTransaction(
+    functionNameOrSignature,
+    args
+  ));
+
+  if (functionEntry.stateMutability !== "view") {
+    console.log(
+      `WARNING!!! Not a view function\n  ABI stateMutability: ${functionEntry.stateMutability}`
+    );
+  }
+
+  const provider = new Encoder.ProviderAdapter(config.provider);
+  let result;
+  try {
+    result = await provider.request({
+      method: "eth_call",
+      params: [
+        {
+          from: config.from,
+          to: transaction.to,
+          data: transaction.data
+        },
+        config.blockNumber
+      ]
+    });
+  } catch (error) {
+    console.log(`Error Message: ${error.message}\nError Code: ${error.code}`);
+  }
+
+  if (result !== null && result !== undefined) {
+    const [decoding] = await decoder.decodeReturnValue(functionEntry, result);
+
+    for (const { value: result } of decoding.arguments) {
```

Comment:
This seemingly doesn't handle different kinds of decodings?  Like, this seems OK for decodings with `kind = "return"`, but there are other kinds as well.  If it's not a kind that has `arguments`, this will error.  Not only that, but if you get back a revert string (on a client that doesn't error on such things), it will print it out as if it were a valid returned result!

One way to simplify this would be to use `Codec.Export.ReturndataDecodingInspector`, since that already handles all these cases.  That said, maybe you don't like how it formats its output. :)  Well -- if that's the case, uh, talk to me and maybe we can come up with something?

Reply:
Btw, for the particular case of panic codes, you may want to also consider making use of the `panicString` function in `debug-utils` to provide a readable interpretation of the panic code to the user.